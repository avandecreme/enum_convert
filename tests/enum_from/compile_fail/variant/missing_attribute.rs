use enum_convert::EnumFrom;

enum Source {
    Foo,
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    // Missing #[enum_from(Source::Foo)]
    Bar,
}

fn main() {}
