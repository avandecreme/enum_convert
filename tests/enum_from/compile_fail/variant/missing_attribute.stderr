error[E0004]: non-exhaustive patterns: `Source::Foo` not covered
 --> tests/enum_from/compile_fail/variant/missing_attribute.rs:7:10
  |
7 | #[derive(EnumFrom)]
  |          ^^^^^^^^ pattern `Source::Foo` not covered
  |
note: `Source` defined here
 --> tests/enum_from/compile_fail/variant/missing_attribute.rs:3:6
  |
3 | enum Source {
  |      ^^^^^^
4 |     Foo,
  |     --- not covered
  = note: the matched value is of type `Source`
  = note: this error originates in the derive macro `EnumFrom` (in Nightly builds, run with -Z macro-backtrace for more info)
